esmvaltool/preprocessor/_derive/__init__.py

```diff
@@ -0,0 +1,118 @@
+"""Automatically derive variables."""
+
+
+import importlib
```

Comment:
Maybe you can just use the variable short name (with the first letter in uppercase of course) as the class name and import everything here with a normal `import` statement? That would be much easier to read than all this importlib usage.

Reply:
I still use `importlib` to dynamically search the `_derive` directory for variables, otherwise I would need to add the import statements manually which is not really convenient in my opinion.